django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,22 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        registered_tables = {
+            model._meta.db_table: model
+            for model in self.opts.apps.get_models(include_auto_created=True)
+        }
+        if self.db_table in set(registered_tables.keys()):
+            msg = 'Field is using a table that has already been registered by {!r}.'
```

Comment:
I would use the model `_meta.app_label` and `object_name` instead of its representation.